src/backend/commands/explain.c

```diff
@@ -1221,64 +1221,15 @@ ExplainNode(PlanState *planstate, List *ancestors,
 	parentplanstate = es->parentPlanState;
 	es->parentPlanState = planstate;
 
-	if (Gp_role == GP_ROLE_DISPATCH)
+	if (save_currentSlice != NULL && save_currentSlice->gangSize > 0)
 	{
-		/*
-		 * Estimates will have to be scaled down to be per-segment (except in a
-		 * few cases).
-		 */
-		if ((plan->directDispatch).isDirectDispatch)
-		{
-			scaleFactor = 1.0;
-		}
-		else if (plan->flow != NULL && CdbPathLocus_IsBottleneck(*(plan->flow)))
-		{
-			/*
-			 * Data is unified in one place (singleQE or QD), or executed on a
-			 * single segment.  We scale up estimates to make it global.  We
-			 * will later amend this for Motion nodes.
-			 */
-			scaleFactor = 1.0;
-		}
-		else if (plan->flow != NULL && CdbPathLocus_IsSegmentGeneral(*(plan->flow)))
-		{
-			/* Replicated table has full data on every segment */
-			scaleFactor = 1.0;
-		}
-		else if (plan->flow != NULL && es->pstmt->planGen == PLANGEN_PLANNER)
-		{
-			/*
-			 * The plan node is executed on multiple nodes, so scale down the
-			 * number of rows seen by each segment
-			 */
-			scaleFactor = CdbPathLocus_NumSegments(*(plan->flow));
-		}
-		else
-		{
-			/*
-			 * The plan node is executed on multiple nodes, so scale down the
-			 * number of rows seen by each segment
-			 */
-			scaleFactor = getgpsegmentCount();
-		}
+		scaleFactor = (float) save_currentSlice->gangSize;
 	}
 
-	/*
-	 * If this is a Motion node, we're descending into a new slice.
-	 */
-	if (IsA(plan, Motion))
+	if (plan->flow != NULL && CdbPathLocus_IsSegmentGeneral(*(plan->flow)))
```

Comment:
Why do we still need flow? ORCA does not fill it correctly!

Reply:
There's still postgres optimizer